pkg/apis/extensions/v1beta1/types.go

```diff
@@ -70,30 +70,49 @@ type SubresourceReference struct {
-	Quantity resource.Quantity `json:"quantity,omitempty"`
+// MetricName is an identifier of a Heapster metric.
+type MetricName string
+
+const (
+	// Heapster metric representing CPU usage of a pod (in milli-cores).
+	MetricCPUUsage MetricName = "cpu-usage"
+	// Heapster metric representing Memory usage of a pod (in bytes).
+	MetricMemoryUsage MetricName = "memory-usage"
+)
+
+// MetricUtilization is an object for specifying the arithmetic mean utilization of a particular metric.
+type MetricUtilization struct {
+	// Name specifies either the identifier of the target metric when present in the spec, or the identifier of the observed metric when present in the status.
+	Name MetricName `json:"name"`
+	// Utilization specifies either the target arithmetic mean utilization of the metric when present in the spec, or the observed arithmetic mean utilization when present in the status.
+	Utilization resource.Quantity `json:"utilization"`
 }
 
 // HorizontalPodAutoscalerSpec is the specification of a horizontal pod autoscaler.
 type HorizontalPodAutoscalerSpec struct {
 	// ScaleRef is a reference to Scale subresource. HorizontalPodAutoscaler will learn the current resource consumption from its status,
 	// and will set the desired number of pods by modyfying its spec.
-	ScaleRef *SubresourceReference `json:"scaleRef"`
-	// MinReplicas is the lower limit for the number of pods that can be set by the autoscaler.
-	MinReplicas int `json:"minReplicas"`
+	ScaleRef SubresourceReference `json:"scaleRef"`
+	// MinReplicas is the lower limit for the number of pods that can be set by the autoscaler, default 1.
+	MinReplicas *int `json:"minReplicas"`
 	// MaxReplicas is the upper limit for the number of pods that can be set by the autoscaler. It cannot be smaller than MinReplicas.
 	MaxReplicas int `json:"maxReplicas"`
-	// Target is the target average consumption of the given resource that the autoscaler will try to maintain by adjusting the desired number of pods.
-	// Currently two types of resources are supported: "cpu" and "memory".
-	Target ResourceConsumption `json:"target"`
+	// TargetMetricUtilizations are the target arithmetic mean utilizations of the latest samles of the given metrics gathered accross pods.
+	// The autoscaler will try to maintain the target by adjusting the desired number of pods.
+	// Currently two types of metrics are supported: "cpu-usage" and "memory-usage".
+	// Currently this array must contain exactly one element.
+	TargetMetricUtilizations []MetricUtilization `json:"targetMetricUtilizations" patchStrategy:"merge" patchMergeKey:"name"`
 }
 
 // HorizontalPodAutoscalerStatus contains the current status of a horizontal pod autoscaler
 type HorizontalPodAutoscalerStatus struct {
+	// ObservedGeneration is the most recent generation observed by this autoscaler.
+	ObserveGeneration *int64 `json:"observedGeneration,omitempty"`
```

Comment:
Go field name and json field name should match except for capitalization of the first letter, and this field should match the one in ReplicationController. ObservedGeneration.